Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -575,7 +575,7 @@ u32 Value::to_u32(GlobalObject& global_object) const
     if (signbit(value))
         int_val = -int_val;
     auto int32bit = fmod(int_val, NumericLimits<u32>::max() + 1.0);
-    return static_cast<u32>(int32bit);
+    return static_cast<u32>(static_cast<i64>(int32bit));
```

Comment:
This looks weird enough that someone might delete it in the future if there's no comment. So I think we need a comment here :)